lib/Sema/TypeCheckNameLookup.cpp

```diff
@@ -296,18 +260,18 @@ LookupResult
 TypeChecker::lookupUnqualifiedType(DeclContext *dc, DeclName name,
                                    SourceLoc loc,
                                    NameLookupOptions options) {
+  auto &ctx = dc->getASTContext();
   auto ulOptions = convertToUnqualifiedLookupOptions(options) |
```

Comment:
Having this mutate the base value makes me nervous.  Can you `const`-qualify this and make copies in the blocks below?

Reply:
Given we can now pass the option set directly, I've removed the mutation.